sahi/model.py

```diff
@@ -65,7 +65,7 @@ def __init__(
         self._object_prediction_list_per_image = None
 
         # automatically set device if its None
-        if not (self.device):
+        if not (self.set_device):
```

Comment:
set_device fonksiyonu her zaman var zaten, var mi diye kontrol etmeye gerek yok.

device set edilmis mi diye kontrol etmemiz gerekiyor.

o yuzden dogrusu `if not self.device:` olmali

Reply:
Zaten önceden self.device değil miydi? Yorum olarak set_device çağrılması gerekiyor dediğiniz için düzelttim. 

Eğer torch device yerine farklı bir kütüphane kullanmak istiyorsak set_device fonksiyonunu kullanmamız gerekiyor. Eğer bu tanımlı değilse torch device kütüphanesi otomatik cpu veya gpu seçsin diye düşünüyorum. Şuan nasıl bir şey tasarlamak istediğinizi anlayamadığımdan kodlama kısmını pek yapamıyorum.

Reply:
Bunun yerine:

```python
self.device = "cuda:0" if is_torch_cuda_available() else "cpu"
```

Boyle olsun istiyoruz:

```python
if self.device is None:
    self.set_device()

Yani device secme islemlerini otomatik olarak set_device() fonksiyonu halledecek bundan sonra.
```

Reply:
Açıklama için teşekkür ederim.